lib/net.js

```diff
@@ -954,7 +954,7 @@ Socket.prototype.connect = function() {
     initSocketHandle(this);
   }
 
-  if (cb !== null) {
+  if (cb) {
```

Comment:
I think we need to check if cb is undefined, to respect API contract: `net.connect(options[,` connectListener])`
I don't think that `net.connect(options, 42)` or `net.connect(options, 'anything')` should silently ignore the second parameter without failure, what do you think ?